bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -71,11 +80,15 @@
     final long ledgerId;
     long lastAddPushed;
     volatile long lastAddConfirmed;
+    volatile long explicitLastAddConfirmed;
+    volatile long piggyBackedLastAddConfirmed;
+
     long length;
     final DigestManager macManager;
     final DistributionSchedule distributionSchedule;
 
     final RateLimiter throttler;
+    private ScheduledExecutorService lastAddConfirmedUpdateScheduler = null;
```

Comment:
@reddycharan 

> 1. 

yes.

> 2. 

we should share the resource usage. mainWorkerPool is fine unless you have other concerns.